test/e2e/tests/request-queuing/multiple-networks-dapps-txs.spec.js

```diff
@@ -0,0 +1,172 @@
+          '[data-testid="account-options-menu-button"]';
+        await driver.waitForSelector(accountOptionsMenuSelector);
+        await driver.clickElement(accountOptionsMenuSelector);
+
+        // Click settings from dropdown menu
+        const globalMenuSettingsSelector =
+          '[data-testid="global-menu-settings"]';
+        await driver.waitForSelector(globalMenuSettingsSelector);
+        await driver.clickElement(globalMenuSettingsSelector);
+
+        // Click Experimental tab
+        const securityAndPrivacyTabRawLocator = {
+          text: 'Experimental',
+          tag: 'div',
+        };
+        await driver.clickElement(securityAndPrivacyTabRawLocator);
+
+        // Toggle request queue setting
+        await driver.clickElement('.request-queue-toggle');
+
+        // Navigate to extension home screen
+        await driver.navigate(PAGES.HOME);
+
+        // Open Dapp One
+        await openDapp(driver, undefined, DAPP_URL);
+
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        // Network Selector
+        await driver.clickElement('[data-testid="network-display"]');
+
+        // Switch to second network
+        await driver.clickElement({
+          text: 'Localhost 8546',
+          css: 'p',
+        });
+
+        // TODO: Request Queuing bug when opening both dapps at the same time will have them stuck on the same network, with will be incorrect for one of them.
+        // Open Dapp Two
+        await openDapp(driver, undefined, DAPP_ONE_URL);
+
+        // Window Handling
+        const windowHandles = await driver.getAllWindowHandles();
+        const dappOne = windowHandles[1];
+        const dappTwo = windowHandles[2];
+
+        // Dapp one send tx
+        await driver.switchToWindow(dappOne);
+        await driver.clickElement('#sendButton');
+
+        // Dapp two send tx
+        await driver.switchToWindow(dappTwo);
+        await driver.clickElement('#sendButton');
+
+        // First switch network confirmation
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+        await driver.clickElement({ text: 'Switch network', tag: 'button' });
+
+        // Wait for confirm tx after switch network confirmation.
+        await driver.delay(regularDelayMs);
+
+        await driver.waitUntilXWindowHandles(4);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        // Find correct network on confirm tx
+        await driver.findElement({
+          text: 'Localhost 8545',
+          tag: 'span',
+        });
+
+        // Reject Transaction
+        await driver.findClickableElement({ text: 'Confirm', tag: 'button' });
+        await driver.clickElement(
+          '[data-testid="page-container-footer-cancel"]',
+        );
+
+        // TODO: No second confirmation from dapp two will show, have to go back to the extension to see the switch chain & dapp two's tx.
```

Comment:
nice find on getting this to work!

Similar to above, it might be good to have a failing test on a branch so we can track this as things are updated /merged